### PR TITLE
FIX: update react-native sample app package files

### PR DESCRIPTION
**Pull request by GGonryun**
This PR should get rid of all of the warnings from dependabot for the react-native-app sample package. The package is outdated and I had a hard time running the android/ios devices so i created https://fullstory.atlassian.net/browse/ECO-8903 to update the sample app to the newest versions.

this pr was verified using `npm run web` to test the web components without the emulators.

Instructions for updating expo: https://docs.expo.dev/workflow/upgrading-expo-sdk-walkthrough/

this PR should fix:

#118
#124
#126
#129
#137
#147
#151
#154

### CODE REVIEW

File: samples/react-native-app/App.js [hunk 1]
```diff
@@ -9,7 +10,9 @@ const instructions = Platform.select({
 export default function App() {
   return (
     <View style={styles.container}>
-      <Text style={styles.welcome}>Welcome to React Native!</Text>
+      <Container>
```
Comment (GGonryun) [inline samples/react-native-app/App.js:13]: technically these samples have been broken for web for a while now. as of react native web 12.0.0 components like "View" and "Text" no longer forward arbitrary props. I created a <Container> element to get around this issue.

https://github.com/necolas/react-native-web/issues/1146

File: samples/react-native-app/package.json [hunk 2]
```diff
@@ -8,26 +8,30 @@
...
+    "@babel/core": "~7.23.2",
     "babel-jest": "~25.2.6",
-    "jest": "^28.1.1",
+    "jest": "^29.2.1",
     "react-test-renderer": "~16.9.0"
   },
   "jest": {
     "preset": "react-native"
   },
+  "overrides": {
```
Comment (GGonryun) [inline samples/react-native-app/package.json:33]: this is the same overrides as what's in the main package, helps us get around the fact that lots of packages that depend on semver haven't been updated.

**Review by djwhatley-fs — APPROVED** [review]
LGTM

**Pull request merged**